### PR TITLE
JUCX: UCP memory registration functionality.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Register memory in JUCX.

## Why ?
To be able to perform RDMA operations in JUCX

## How ?
Added to `UcpContext` 2 public methods: `allocateMemory` and `registerMemory` to allocate or register `MemoryMappedBuffer`.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9862/ for details (Mellanox internal link).

File: bindings/java/src/main/java/org/ucx/jucx/ucp/UcpContext.java [hunk 1]
```diff
@@ -41,8 +44,37 @@ public void close() {
...
+     * with it. The network stack associated with an application context
+     * can typically send and receive data from the mapped memory without
+     * CPU intervention; some devices and associated network stacks
+     * require the memory to be mapped to send and receive data.
+     *
+     * @param size - since java uses int for array indexes,
+     *               maximum memory allocation size is limited to 2Gb
+     * @return
+     */
+    public UcpMemory allocateMemory(int size) {
```
Comment (yosefe) [inline bindings/java/src/main/java/org/ucx/jucx/ucp/UcpContext.java]: why do we need to have memory allocation API ? isnt it enough to have registration?

File: bindings/java/src/main/java/org/ucx/jucx/ucp/UcpMemory.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     * This routine packs the RKEY object in a portable format such that the
+     * object can be "unpacked" on any platform supported by the
+     * UCP library.
+     * RKEYs for InfiniBand and Cray Aries networks typically includes
+     * InifiniBand and Aries key.
+     * In order to enable remote direct memory access to the memory associated
+     * with the memory handle the application is responsible for sharing the RKEY with
+     * the peers that will initiate the access.
+     */
+    public ByteBuffer getrKeyBuffer() {
```
Comment (yosefe) [inline bindings/java/src/main/java/org/ucx/jucx/ucp/UcpMemory.java]: getRemoteKeyBuffer

File: bindings/java/src/test/java/org/ucx/jucx/UcpMemoryTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+import static java.nio.file.StandardOpenOption.*;
+import static org.junit.Assert.*;
+
+public class UcpMemoryTest {
```
Comment (yosefe) [inline bindings/java/src/test/java/org/ucx/jucx/UcpMemoryTest.java]: is this test going to run in jenkins?

Comment (petro-rudenko) [reply bindings/java/src/test/java/org/ucx/jucx/UcpMemoryTest.java]: Yes

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9872/ for details (Mellanox internal link).

**Pull request merged**